src/markov/markov_approx.jl

```diff
@@ -130,3 +130,50 @@ function _rouwenhorst(p::Real, q::Real, m::Real, Δ::Real, n::Integer)
         return linspace(m-Δ, m+Δ, n), θN
     end
 end
+
+
+"""
+Accepts the simulation of a discrete state Markov chain and estimates
+the transition probabilities via maximum likelihood.
+
+Note: Only states that are observed in the history will be included
+in the estimation.
+
+##### Arguments
+
+- `X::Vector{T}` : Simulated history of Markov states
+
+##### Returns
+
+- `mc::MarkovChain{T}` : A Markov chain holding the state values and
+transition matrix
+
+"""
+function estimate_MC_discrete{T}(X::Vector{T})
+    # Get length of simulation
+    capT = length(X)
+
+    # Find all unique observations and sort them.
+    states = sort!(unique(X))
+    nstates = length(states)
+
+    # Counter matrix
+    cm = zeros(nstates, nstates)
+
+    # Compute conditional probabilities for each state
+    state_i = findfirst(states, X[1])
+    for t in 1:capT-1
+        # Find next period's state
+        state_j = findfirst(states, X[t+1])
```

Comment:
I could use `searchsortedfirst` which "Returns the index of the first value in `a` greater than or equal to `x`." I preferred `findfirst` because it is only checking for equality. This means that if it doesn't find the value then the index returned will be 0 and it will throw an error when it tries to use it... I'd prefer it to throw this error than silently just take the next largest value.

Unless the number of states you have is very big then I suspect this isn't a relevant performance hit -- And if you have that many states then you should be worried about having a long enough simulation to generate accurate transition matrices.